etna/search/templatetags/search_tags.py

```diff
@@ -137,3 +131,19 @@ def extended_in_operator(lhs_operand, *rhs_operand_list) -> bool:
     Returns True when rhs_operand_list contains lhs_operand value, False otherwise
     """
     return (lhs_operand in rhs_operand_list) or False
+
+
+@register.filter
+def non_field_error_by_code(non_field_error_code, form) -> str:
+    """
+    Returns non field error html for the error code.
+    """
+    for item in form.non_field_errors().data:
+        if item.code == non_field_error_code:
+            error_text = item.error_list[0].message
+            html = '<ul class="errorlist nonfield"><li>{error_text}</li></ul>'.format(
+                error_text=error_text
+            )
+            return mark_safe(html)
+
+    return ""
```

Comment:
@TNA-Allan I'm afraid I'm not sure where/how this relates to the other changes... could we perhaps save this for a separate PR?

Reply:
> Hi @TNA-Allan, I've done what I can with review here. The general gist is: This additional complexity has highlighted just how fragile and inflexible the current form rendering is, and we need to take a step back and think about adjusting things properly to make things more dynamic, and have 'configuration' in the right place.

Agree.

> A custom filter just isn't the right device here. This should really be [simple tag](https://docs.djangoproject.com/en/4.2/howto/custom-template-tags/#simple-tags), and we need to find a way for these values to be specified on the form instance, so that they aren't hardcoded in the template

Perhaps would it be bound field methods, but I'm not sure how to go about it with the complexities for this specific page with multiple forms and their behaviours (and upcoming tickets to reflect that).

Reply:
Some how, I've lost this comment and add it again - may have linked to some other comment below while I tried to delete it.

Reply:
@ababic I don't understand the idea of a subfield